tajo-plan/src/main/java/org/apache/tajo/plan/joinorder/GreedyHeuristicJoinOrderAlgorithm.java

```diff
@@ -57,17 +54,69 @@ public FoundJoinOrder findBestOrder(LogicalPlan plan, LogicalPlan.QueryBlock blo
     JoinEdge bestPair;
 
     while (remainRelations.size() > 1) {
+      Set<LogicalNode> checkingRelations = new LinkedHashSet<LogicalNode>();
+
+      for (LogicalNode relation : remainRelations) {
+        Collection <String> relationStrings = PlannerUtil.getRelationLineageWithinQueryBlock(plan, relation);
+        List<JoinEdge> joinEdges = new ArrayList<JoinEdge>();
+        String relationCollection = TUtil.collectionToString(relationStrings, ",");
+        List<JoinEdge> joinEdgesForGiven = joinGraph.getIncomingEdges(relationCollection);
+        if (joinEdgesForGiven != null) {
+          joinEdges.addAll(joinEdgesForGiven);
+        }
+        for (String relationString: relationStrings) {
```

Comment:
Would you mind explaning these lines?
I found that the same join edge is added twice to the ```joinEdgesForGiven``` list.